activerecord/lib/active_record/reflection.rb

```diff
@@ -644,11 +644,14 @@ def valid_inverse_reflection?(reflection)
         #
         # Anything with a scope can additionally ruin our attempt at finding an
         # inverse, so we exclude reflections with scopes.
+        #
+        # We should also ignore polymorphic associations.
         def can_find_inverse_of_automatically?(reflection)
           reflection.options[:inverse_of] != false &&
             VALID_AUTOMATIC_INVERSE_MACROS.include?(reflection.macro) &&
             !INVALID_AUTOMATIC_INVERSE_OPTIONS.any? { |opt| reflection.options[opt] } &&
-            !reflection.scope
+            !reflection.scope &&
+            !polymorphic?
```

Comment:
This issue was caused by #28808 due to removing `:polymorphic` in `INVALID_AUTOMATIC_INVERSE_OPTIONS` to allow that `has_one` and `has_many` associations can find a polymorphic inverse association (`can_find_inverse_of_automatically?` is also used for validating found inverse reflection).

#28808 intend to remove checking like `reflection.polymorphic?` to found inverse reflection, but still can not find inverse association if `self` is `belongs_to` polymorphic reflection.

Can you place this change under `BelongsToReflection` to make the effect/intent of this easier to understand?

```diff
diff --git a/activerecord/lib/active_record/reflection.rb b/activerecord/lib/active_record/reflection.rb
index e640d75d2f..3342cb77ee 100644
--- a/activerecord/lib/active_record/reflection.rb
+++ b/activerecord/lib/active_record/reflection.rb
@@ -732,6 +732,9 @@ def join_foreign_key
       end
 
       private
+        def can_find_inverse_of_automatically?(_)
+          !polymorphic? && super
+        end
 
         def calculate_constructable(macro, options)
           !polymorphic?
```